app/Models/Image.php

```diff
@@ -191,4 +193,23 @@ public function ensureDirectoryExists()
             mkdir($destination, 0777, true);
         }
     }
+
+    /**
+     * Deletes any generated files if they exist
+     * @param bool $includeOriginal Deletes
+     */
+    public function clearExisting($includeOriginal = false) {
+        $files = scandir($this->getDirectory());
+        $filePrefix = $this->id.'_';
+        $originalName = $filePrefix.Image::$ImageTypes[Image::ORIGINAL]['name'];
+
+        $files = array_filter($files, function($file) use ($originalName, $includeOriginal, $filePrefix) {
+            if (Str::startsWith($file,$originalName) && !$includeOriginal) return false;
```

Comment:
Pony.fm's codebase does not use shorthand `if` syntax; you should generally always write out curly braces to separate an if statement's cases. This reduces the chance of errors when modifying the code in the future.